### DISCUSSION TITLE
Running stackSentinel.py without bbox

### DISCUSSION THREAD

**Discussion by jaha2600** [isce-framework/isce2#463] [Q&A] [answered]
Is it possible to run stackSentinel.py without the use of a -b flag?

It is not listed as a Required Parameter in the [topsStack](https://github.com/isce-framework/isce2/tree/main/contrib/stack/topsStack) readme file, however stackSentinel.py fails without it.

I am looking to run a stack of SLC files where I process the whole extent, as opposed to a bounding box subset

thanks!

**Comment by hfattahi** [comment, marked as answer]
It should have been listed as required. In practice as you saw, it is required anyways. The reason for that is the fact that frames of Sentinel-1 are not fixed on ground. Therefore best practice is that user should know what area they want to process. The stack processor will still check all the SLCs and will provide some info on what common region exists. Keep in mind, because of the coverage issue, you may always download data for a bounding box larger than the bounding box you give to stackSentinel for processing (around 0.25 degrees larger to be safe), so that you will get the coverage that you want.

**Reply by jaha2600** [reply]
Thanks for the reply - this makes sense.

Even when choosing a bounding box that covers all of the SLC files (using the provided common region) I often get errors about there being a different number of bursts in images, this appears on any step that attempts to merge (merge_ref_sec, merge_burst_igram). Do you happen to know why I might be seeing this error? 

`ValueError: /data/GREENLAND/2022/ISCE_2/interferograms/20210912_20210924 has different number of bursts (8) than the reference (9)`

**Reply by tryingphysics000** [reply]
> Thanks for the reply - this makes sense.
> 
> Even when choosing a bounding box that covers all of the SLC files (using the provided common region) I often get errors about there being a different number of bursts in images, this appears on any step that attempts to merge (merge_ref_sec, merge_burst_igram). Do you happen to know why I might be seeing this error?
> 
> `ValueError: /data/GREENLAND/2022/ISCE_2/interferograms/20210912_20210924 has different number of bursts (8) than the reference (9)`

Hi @jaha2600 ,

Have you the figured out the underlying cause of different number of bursts? I am facing the same issue.

Thank you.

**Comment by yuluenter** [comment]
Hello @hfattahi  , I would like to know if there is a way to subset SLCs generated by stackSentinel.py to a smaller size.  I know that every SLC will eventually contain at least two bursts, but my study area is too small, and I hope to further narrow the area before the StaMPS processing.

**Reply by nadiahassanpour** [reply]
Dear Yuluenter, it is my question too, Did you get your answer or not?

**Reply by yuluenter** [reply]
I have solved this problem. You should be able to find "run_SLCcropStack.csh" in the prepStackToStaMPS/bin directory of ISCE2. then copy this file to your project folder, change the contents of  "data_path", "proc_dir", and "crop_rdr.py -b" in this file, and run it!

**Reply by nadiahassanpour** [reply]
Thanks for your answer, I should say that I want to run mintpy after ISCE. Do It work for isce 2 mintpy process?

**Reply by yuluenter** [reply]
As far as I know mintpy only accepts unwrapped phase. So the -W parameter of stackSentinel needs to be set to "interferogram". This will take a long time in the phase unwrapping step of stackSentinel, and I haven't found an efficient way to handle it yet.

**Reply by nadiahassanpour** [reply]
Yes, it takes a long time. thanks a lot, dear Yuluenter. just one more question, Did you work with TSX before?
Did you know how can I run the files (run_0* and .job) in the runfiles folder?

**Reply by yuluenter** [reply]
Perhaps a script could be written specifically for cropping interferograms. And run this script before the last step of phase unwrapping. Actually, "run_SLCcropStack.csh" just calls some tools of GDAL, which is not too complicated. 
And I haven't worked with TSX data, sorry I can't help.

**Reply by nadiahassanpour** [reply]
you're right thanks a lot.
